doc/source/data/transforming-datasets.rst

```diff
@@ -130,16 +130,26 @@ Here is an overview of the available batch formats:
     This may incur a conversion cost if the underlying Dataset block is not
     zero-copy convertible from an Arrow table.
 
+  .. literalinclude:: ./doc_code/transforming_datasets.py
+    :language: python
+    :start-after: __writing_native_udfs_tabular_begin__
+    :end-before: __writing_native_udfs_tabular_end__
+
   * **Tensor Datasets** (single-column): Each batch will be a single
     `numpy.ndarray <https://numpy.org/doc/stable/reference/generated/numpy.ndarray.html>`__
     containing the single tensor column for this batch.
 
-  * **Simple Datasets**: Each batch will be a Python list.
+  .. literalinclude:: ./doc_code/transforming_datasets.py
+    :language: python
+    :start-after: __writing_native_udfs_tensor_begin__
+    :end-before: __writing_native_udfs_tensor_end__
+
+  * **List Datasets**: Each batch will be a Python list.
```

Comment:
This seems fine. However in other places, we still call it "Simple Datasets". I think we'd keep it consistent.

Reply:
If we're going to make this change, we should change the naming to "List Datasets" everywhere, in a single PR, so we keep this consistent.

Reply:
Got it. Leave it as Simple Datasets for consistency.